pkg/kubelet/prober/worker.go

```diff
@@ -208,7 +208,9 @@ func (w *worker) doProbe() (keepGoing bool) {
 			w.resultsManager.Remove(w.containerID)
 		}
 		w.containerID = kubecontainer.ParseContainerID(c.ContainerID)
-		w.resultsManager.Set(w.containerID, w.initialValue, w.pod)
+		if c.State.Running == nil || int32(time.Since(c.State.Running.StartedAt.Time).Seconds()) < w.spec.InitialDelaySeconds || w.probeType == startup || w.probeType == liveness {
```

Comment:
Can you just explain the logic, and maybe put a comment or extract the if to clearly understand when we should (not) set the initial value?

Reply:
Problem is that these lines of code are always executed nomatter what. Reason is that w.containerID is not filled upon creation of worker but only here by copying from c.ContainerID

So there needed to be some condition upon which we actually set the initialValue. This PR aims to improve readiness check so we exclude other probeTypes. It might be worth improving this worker.go in general by creating seperate worker_types based on their proberType with their own coding.

The other conditions cover truly new containers which are started for first time, they can be set to initialValue as it is no harm for them.

In general, upon first run worker.go will call resultmanager twice today. Once for filling w.ContainerID and another time for the actual readiness check. I wanted to get rid of the first call because it is stupid to call resultManager with a value which was not dervied from actual testing.

If desired, i can add a comment.